src/test/java/com/puppycrawl/tools/checkstyle/bdd/TestInputConfiguration.java

```diff
@@ -68,12 +70,16 @@ public final class TestInputConfiguration {
 
     private final List<TestInputViolation> filteredViolations;
 
+    private final Configuration xmlConfiguration;
+
     private TestInputConfiguration(List<ModuleInputConfiguration> childrenModules,
                                    List<TestInputViolation> violations,
-                                   List<TestInputViolation> filteredViolations) {
+                                   List<TestInputViolation> filteredViolations,
+                                   Configuration xmlConfiguration) {
```

Comment:
If we have xml, you do not need `(List<ModuleInputConfiguration> childrenModules`
Create new ctor